src/model/entities/core-device.ts

```diff
@@ -0,0 +1,9 @@
+import { StatusCode } from '../enums/status-code'
+
+export interface CoreDevice {
```

Comment:
This interface shouldn't exist.
It shares all the same values as `Device`, so we should simply just use `Device`.